File: test17_out/pollen.events/Dispatcher/Dispatcher.js

/*
 * ======== MODULE pollen.events.Dispatcher ========
 */

var $$c = function() {
    const Dispatcher = {};
    Dispatcher.Dispatcher = Dispatcher
    Dispatcher.$name = 'pollen.events.Dispatcher';
    Dispatcher.pollen$used = 0;
    Dispatcher.AE = $units['pollen.events.AE'];
    Dispatcher.AEQueue = $units['pollen.data.AEQueue'];
    Dispatcher.post = new $$Ref('pollen_events_Dispatcher_post__E');
    Dispatcher.postFromInterrupt = new $$Ref('pollen_events_Dispatcher_postFromInterrupt__E');
    Dispatcher.run = new $$Ref('pollen_events_Dispatcher_run__E');
    Dispatcher.targetInit = new $$Ref('pollen_events_Dispatcher_targetInit__I');
    Dispatcher.bindGlobalInterrupts = function( g ) {
        var $$text = '';
        
        return $$text;
    }
    Dispatcher.$$hostInit = function() {
        var $$text = '';
        return $$text;
    }
    Dispatcher.pollen__uses__ = function() {
        $$bind($units['pollen.events.AE'], 'pollen$used', true);
        $$bind($units['pollen.data.AEQueue'], 'pollen$used', true);
    }
    Dispatcher.$$privateInit = function() {
    }
    return Dispatcher;
}
var $$u = $$c();
$units['pollen.events.Dispatcher'] = $$u;
$units.push($$u);

File: test17_out/pollen.events/Dispatcher/Dispatcher.c

/*
 * ======== string literals ========
 */

static const char pollen_events_Dispatcher__s__fileName[] = "Dispatcher.p";
static const char pollen_events_Dispatcher__s__0[] = "Dispatcher";

/*
 * ======== function definitions ========
 */

void pollen_events_Dispatcher_postFromInterrupt__E( struct pollen_events_AE*  e ) {
    pollen_data_AEQueue_AEQueue_enqueue__E(Dispatcher.AEQueue, (e));
}

void pollen_events_Dispatcher_post__E( struct pollen_events_AE*  e ) {
    bool on;
    on = pollen_interfaces_GlobalInterrupts_disable__E();
    pollen_data_AEQueue_AEQueue_enqueue__E(Dispatcher.AEQueue, (e));
    pollen_interfaces_GlobalInterrupts_restore__E((on));
}

void pollen_events_Dispatcher_run__E() {
    bool on;
    pollen_events_AE_ e;
    pollen_interfaces_GlobalInterrupts_enable__E();
    while (true) {
        on = pollen_interfaces_GlobalInterrupts_disable__E();
        e = pollen_data_AEQueue_AEQueue_dequeue__E(Dispatcher.AEQueue);
        pollen_interfaces_GlobalInterrupts_restore__E((on));
        if ((e) != null) {
            pollen_events_AE_AE_handle__E(e);
        }
    }
}

void pollen_events_Dispatcher_targetInit__I() {
}

